src/App.js

```diff
@@ -83,6 +86,28 @@ function App() {
     });
   }
 
+  const handleModalOpen = (item, itemId) => {
+    // set item to be deleted to item object and set itemId since it's separate from Firestore
+    setItemToDelete({ ...item, id: itemId });
+    setShowModal(true);
+  };
+
+  const handleModalClose = () => {
+    setShowModal(false);
+  };
+
+  const deleteItem = () => {
+    db.collection(`lists/${listId}/items`)
+      .doc(itemToDelete.id)
+      .delete()
+      .then(() => {
+        handleModalClose();
+      })
+      .catch((err) => {
+        console.error('Error removing document: ', err);
+      });
+  };
+
```

Comment:
See my other comment on Modal.js about event listeners getting added too many times. I suggested there that one potential fix would be to split the handleKeyEvents function up, connecting the tab/tab+shift handling to onKeyDown events on the buttons, and then moving the escape key stuff into a separate function here in App.js. Why move it to App.js? So we can wrap it in a useCallback with no dependencies and ensure we only have one version of the function. (In Modal.js handleModalClose would be a dependency because it's passed down as a prop.)

I put sample code for the first part in my other comment, and here's the escape key portion I tested if you want to try it out. (Note that you would need to import useCallback at the top of this file, so `import { useState, useEffect, useCallback } from 'react';`.)

```
  // close modal if user hits Escape (27)
  const closeModalOnEsc = useCallback((e) => {
    if (e.keyCode === 27) handleModalClose();
  }, []);

  useEffect(() => {
    if(showModal) {
      document.addEventListener('keydown', closeModalOnEsc);
    } else {
      document.removeEventListener('keydown', closeModalOnEsc);
    }
  }, [showModal, closeModalOnEsc]);
```
Optional addition: we could even put in the "close when backdrop clicked" feature we didn't have time to finish up pretty easily by expanding on the above code like so...

```
  // close modal if user hits Escape (27)
  const closeModalOnEsc = useCallback((e) => {
    if (e.keyCode === 27) handleModalClose();
  }, []);

  // close modal if user hits Escape (27)
  const closeModalOnClickAway = useCallback((e) => {
    if (e.target.classList.contains('dialog-backdrop')) handleModalClose();
  }, []);

  // when modal opens/closes, add/remove event listeners
  useEffect(() => {
    if(showModal) {
      document.addEventListener('keydown', closeModalOnEsc);
      document.addEventListener('click', closeModalOnClickAway);
    } else {
      document.removeEventListener('keydown', closeModalOnEsc);
      document.removeEventListener('click', closeModalOnClickAway);
    }
  }, [showModal, closeModalOnEsc, closeModalOnClickAway]);
```